webpack/transformers/descriptionTransformer.js

```diff
@@ -0,0 +1,270 @@
+ * supply it as the argument for the description decorator.
+ */
+class DescriptionTransformer {
+    /**
+     *
+     * @param {ts.TransformationContext} context The transformation context.
+     */
+    constructor(context) {
+        this.context = context;
+    }
+
+    /**
+     * @type {ts.TransformationContext} The transformation context.
+     */
+    context;
+
+    /**
+     * Set to `true` if the file processed by this transformer is an IDE file.
+     */
+    isIDEFile = false;
+
+    /**
+     * Checks whether the given node has a decorator or decorator factory with the given name.
+     * @param {string} name         The name of the decorator to find.
+     * @param {ts.Node} node        The node in which to search.
+     * @return {boolean}            `true` if the decorator was found, `false` otherwise.
+     */
+    hasDecoratorNamed(name, node) {
+        if (!node.decorators) return false;
+
+        // Getting the decorator name depends on whether the decorator is applied directly or via a
+        // decorator factory.
+        for (const decorator of node.decorators) {
+            // In a decorator factory, the decorator itself is the result of invoking
+            // the decorator factory function so it doesn't technically have a name; in this case the name
+            // of the decorator factory function is considered to be the decorator name.
+            if (decorator.expression.kind == ts.SyntaxKind.CallExpression) {
+                /** @type {ts.CallExpression} */ const callExpression = decorator.expression;
+                if (
+                    callExpression.expression.kind == ts.SyntaxKind.Identifier &&
+                    callExpression.expression.text == name
+                ) {
+                    return true;
+                }
+            } else if (decorator.expression.kind == ts.SyntaxKind.Identifier) {
+                /** @type {ts.Identifier} */ const identifierExpression = decorator.expression;
+                if (identifierExpression.text == name) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Visits the given node. This method will be invoked for all nodes in the file.
+     * @param {ts.Node} node        The node to visit.
+     * @return {ts.Node}            The visited node, or a new node that will replace it.
+     */
+    visit(node) {
+        // The description decorator only makes sense for IDE files
+        // An IDE file is identified from its imports - it must import its Thingworx specific
+        // decorators from the widgetIDESupport package
+        if (node.kind == ts.SyntaxKind.ImportDeclaration) {
+            /** @type {ts.ImportDeclaration} */ const importNode = node;
+
+            /** @type {ts.StringLiteral} */ const module = importNode.moduleSpecifier;
+            if (module.text == 'typescriptwebpacksupport/widgetIDESupport') {
```

Comment:
Just a thought, but would it make sense to look at the inheritance chain of the typescript class and see if it extends from a class called `TWComposerWidget`?

On second thought it won't make much sense since it just increases complexity for no reason, so it's better in the current format